docs/java-rest/low-level/configuration.asciidoc

```diff
@@ -99,3 +99,30 @@ http://docs.oracle.com/javase/8/docs/technotes/guides/net/properties.html[`netwo
 to your
 http://docs.oracle.com/javase/8/docs/technotes/guides/security/PolicyFiles.html[Java
 security policy].
+
+=== Node selector
+
+The client sends each request to one of the configured nodes in round-robin
+fashion. Nodes can optionally be filtered through a node selector that needs
+to be provided when initializing the client. This is useful when sniffing is
+enabled, in case only dedicated master nodes should be hit by HTTP requests.
+For each request the client will run the eventually configured node selector
+to filter the node candidates, then select the next one in the list out of the
+remaining ones.
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/RestClientDocumentation.java[rest-client-init-allocation-aware-selector]
+--------------------------------------------------
+<1> Set an allocation aware node selector that allows to pick a node in the
+local rack if any available, otherwise go to any other node in any rack. It
+acts as a preference rather than a strict requirement, given that it goes to
+another rack if none of the local nodes are available, rather than returning
+no nodes in such case which would make the client forcibly revive a local node
+whenever none of the nodes from the preferred rack is available.
+
+WARNING: Node selectors that select a different set of nodes out of different
```

Comment:
Maybe "NodeSelectors that do not consistently select the same set of nodes will make round-robin...."

Reply:
++ thanks